auth/tests/unit/attestation-cloudwatch-alarms.test.ts

```diff
@@ -0,0 +1,345 @@
+      );
+      expect(cloudWatchAlarmUnderTest.Properties.Namespace).toEqual(
+        "AWS/ApiGateway"
+      );
+
+      expect(cloudWatchAlarmUnderTest.Properties.Statistic).toEqual(statistic);
+      expect(cloudWatchAlarmUnderTest.Properties.ExtendedStatistic).toEqual(
+        extendedStatistic
+      );
+      expect(cloudWatchAlarmUnderTest.Properties.Period).toEqual(period);
+      expect(cloudWatchAlarmUnderTest.Properties.EvaluationPeriods).toEqual(
+        evaluationPeriods
+      );
+      expect(cloudWatchAlarmUnderTest.Properties.DatapointsToAlarm).toEqual(
+        datapointsToAlarm
+      );
+      expect(cloudWatchAlarmUnderTest.Properties.Threshold).toEqual(threshold);
+      expect(cloudWatchAlarmUnderTest.Properties.ComparisonOperator).toEqual(
+        comparisonOperator
+      );
+      expect(cloudWatchAlarmUnderTest.Properties.Dimensions).toEqual(
+        dimensions
+      );
+      expect(cloudWatchAlarmUnderTest.Properties.AlarmActions).toEqual([
+        { Ref: topicResource },
+      ]);
+      expect(cloudWatchAlarmUnderTest.Properties.OKActions).toEqual([
+        { Ref: topicResource },
+      ]);
+      expect(cloudWatchAlarmUnderTest.Properties.Tags).toEqual([
+        { Key: "Product", Value: "GOV.UK" },
+        { Key: "Environment", Value: { Ref: "Environment" } },
+        { Key: "System", Value: "Authentication" },
+      ]);
+    });
+
+    it(`should create a SNS topic for ${metricName}`, () => {
+      expect(snsTopicUnderTest).toBeDefined();
+      expect(snsTopicUnderTest.Type).toEqual("AWS::SNS::Topic");
+      expect(snsTopicUnderTest.Properties).toBeDefined();
+      expect(snsTopicUnderTest.Properties.DisplayName).toEqual({
+        "Fn::Join": [
+          "-",
+          [
+            { Ref: "AWS::StackName" },
+            topicDisplayName,
+            {
+              "Fn::Select": [
+                4,
+                {
+                  "Fn::Split": [
+                    "-",
+                    {
+                      "Fn::Select": [
+                        2,
+                        {
+                          "Fn::Split": ["/", { Ref: "AWS::StackId" }],
+                        },
+                      ],
+                    },
+                  ],
+                },
+              ],
+            },
+          ],
+        ],
+      });
+      expect(snsTopicUnderTest.Properties.KmsMasterKeyId).toEqual({
+        Ref: "CloudWatchAlarmNotificationTopicKey",
+      });
+      expect(snsTopicUnderTest.Properties.Tags).toEqual([
+        { Key: "Product", Value: "GOV.UK" },
+        { Key: "Environment", Value: { Ref: "Environment" } },
+        { Key: "System", Value: "Authentication" },
+      ]);
+    });
+
+    it(`should create a SNS subscription for ${metricName}`, () => {
+      expect(subscriptionUnderTest).toBeDefined();
+      expect(subscriptionUnderTest.Type).toEqual("AWS::SNS::Subscription");
+      expect(subscriptionUnderTest.Properties).toBeDefined();
```

Comment:
Question - do we add the `.toBeDefined();` onto a property that we later go on to test to avoid any type-checking issues?